packages/autorest.python/autorest/codegen/models/property.py

```diff
@@ -130,31 +130,47 @@ def validation(self) -> Optional[Dict[str, Any]]:
         return retval or None
 
     @staticmethod
-    def contain_model_type(t: BaseType) -> bool:
-        from . import ListType, DictionaryType, ModelType
+    def contain_target(t: BaseType, check_target: Callable[[BaseType], bool]) -> bool:
+        from . import ListType, DictionaryType, CombinedType
 
-        if isinstance(t, ModelType):
+        if check_target(t):
             return True
         if isinstance(t, ListType):
-            return Property.contain_model_type(t.element_type)
+            return Property.contain_target(t.element_type, check_target)
```

Comment:
we should do this recursively, i.e. move this onto the types. This way we can just do `self.type.contain_target` (like `self.type.type_definition` for example)

Reply:
Yes, added contain_target as a member function of BaseType: https://github.com/Azure/autorest.python/pull/1733/commits/46c2a68a9b9b3e9c8d46ac40a1a5bd52d2844cc6